designs/discontinue_usage_of_kube_rbac_proxy.md

```diff
@@ -0,0 +1,236 @@
+aligning with their domain of responsibility.
+- **Flexible Network Policy Usage**: Allow users to opt-out of the default-enabled usage of [Network Policies][k8s-doc-networkpolicies]
+if they prefer another solution, plan to deploy their solution with a vendor, or use a CNI that does not support NetworkPolicies.
+
+### Non-Goals
+
+- **Replicate kube-rbac-proxy Features or Protection Level**: It is not a goal to provide exactly the same features
+or layer of protection as [kube-rbac-proxy][kube-rbac-proxy]. Since [Network Policies][k8s-doc-networkpolicies]operate differently
+and do not offer the same kind of functionality as [kube-rbac-proxy][kube-rbac-proxy], achieving identical protection levels through
+[Network Policies][k8s-doc-networkpolicies]alone is not feasible.
+
+However, incorporating NetworkPolicies, cert-manager, and/or the features introduced
+in the [controller-runtime pull request #2407][cr-pr] can potentially address some of the security concerns that
+kube-rbac-proxy handles.
+
+## Proposal
+
+### Phase 1: Transition to NetworkPolicies
+
+The immediate action outlined in this proposal is the replacement of [kube-rbac-proxy][kube-rbac-proxy]
+with Kubernetes API NetworkPolicies.
+
+### Phase 2: Subsequent Enhancements: Cert-Manager and/or Controller-Runtime Integration
+
+Looking beyond the initial phase, this proposal envisions integrating cert-manager for TLS certificate management
+and exploring synergies with new features in Controller Runtime, as demonstrated in [PR #2407](https://github.com/kubernetes-sigs/controller-runtime/pull/2407).
+
+These enhancements would introduce encrypted communication for metrics endpoints and potentially incorporate authentication mechanisms,
+significantly elevating the security model employed by projects scaffolded by Kubebuilder.
+
+- **cert-manager**: Automates the management and issuance of TLS certificates, facilitating encrypted communication and, when configured with mTLS, adding a layer of authentication.
+  Currently, we leverage on cert-manager when webhooks are scaffold. So, the proposal idea would be to allow users enable the cert-manager for the metrics such as it is provided
+  and required for webhook feature. However, it MUST be optional. One of the goals of Kubebuilder is make it easier for new users, therefore new users should
+  not need to deal with cert-manager by default or have the need to install it to just quick start.
+
+- **Controller-Runtime Feature (PR #2407)[cr-pr]**:  introduces options for metrics that allow serving metrics via TLS and using authentication and authorization.
+It seems fit very well to our requirements.
+
+While this setup does not replicate [kube-rbac-proxy's][kube-rbac-proxy] method of leveraging Kubernetes RBAC for authentication
+and authorization directly, it achieves a comparable level of security through a combination of network isolation and encrypted communication.
```

Comment:
I think this setup _does_ do the same thing as kube-rbac-proxy. See https://github.com/kubernetes-sigs/controller-runtime/blob/cf69be219f050a456a93949d0c53773836f9216c/pkg/metrics/filters/filters.go#L29-L32

Reply:
It does!

Reply:
Closing since it is addressed in the proposal as well.